account_manager/src/validator/deposit.rs

```diff
@@ -167,39 +182,79 @@ pub fn cli_run<T: EthSpec>(
         return Err("Refusing to deposit to the zero address. Check testnet configuration.".into());
     }
 
-    let (_event_loop_handle, transport) =
-        Ipc::new(eth1_ipc_path).map_err(|e| format!("Unable to connect to eth1 IPC: {:?}", e))?;
-    let web3 = Web3::new(transport);
-
-    let deposits_fut = async {
-        poll_until_synced(web3.clone(), log.clone()).await?;
-
-        for (mut validator_dir, eth1_deposit_data) in eth1_deposit_datas {
-            let tx_hash = web3
-                .eth()
-                .send_transaction(TransactionRequest {
-                    from: from_address,
-                    to: Some(deposit_contract),
-                    gas: Some(DEPOSIT_GAS.into()),
-                    gas_price: None,
-                    value: Some(from_gwei(eth1_deposit_data.deposit_data.amount)),
-                    data: Some(eth1_deposit_data.rlp.into()),
-                    nonce: None,
-                    condition: None,
-                })
-                .compat()
-                .await
-                .map_err(|e| format!("Failed to send transaction: {:?}", e))?;
-
-            validator_dir
-                .save_eth1_deposit_tx_hash(&format!("{:?}", tx_hash))
-                .map_err(|e| format!("Failed to save tx hash {:?} to disk: {:?}", tx_hash, e))?;
-        }
+    if eth1_rpc_url.len() > 0 {
+        let (_event_loop_handle, http_transport) = Http::new(eth1_rpc_url.as_str())
+            .map_err(|e| format!("Unable to connect to eth1 RPC: {:?}", e))?;
+        let web3_http = Web3::new(http_transport);
+
+        let deposits_fut = async {
+            poll_until_synced(web3_http.clone(), log.clone()).await?;
+
+            for (mut validator_dir, eth1_deposit_data) in eth1_deposit_datas {
```

Comment:
I think we should avoid this code duplication, it's likely to end up getting inconsistent over time.

I think we can do it by moving L188-L2220 into a function that's generic over `web::Transport`.

Reply:
done!